run/service-auth/receive_test.py

```diff
@@ -25,33 +25,55 @@
 from requests.adapters import HTTPAdapter
 from requests.packages.urllib3.util.retry import Retry
 
-
-@pytest.fixture()
-def services():
-    # Unique suffix to create distinct service names
-    suffix = uuid.uuid4().hex
-    service_name = f"receive-{suffix}"
-    project = os.environ["GOOGLE_CLOUD_PROJECT"]
-
-    # Deploy receive Cloud Run Service
+PROJECT_ID = os.environ["GOOGLE_CLOUD_PROJECT"]
+
+HTTP_STATUS_OK = 200
+HTTP_STATUS_BAD_REQUEST = 400
+HTTP_STATUS_UNAUTHORIZED = 401
+HTTP_STATUS_FORBIDDEN = 403
+HTTP_STATUS_NOT_FOUND = 404
+HTTP_STATUS_INTERNAL_SERVER_ERROR = 500
+HTTP_STATUS_BAD_GATEWAY = 502
+HTTP_STATUS_SERVICE_UNAVAILABLE = 503
+HTTP_STATUS_GATEWAY_TIMEOUT = 504
+
+STATUS_FORCELIST = [
+    HTTP_STATUS_BAD_REQUEST,
+    HTTP_STATUS_UNAUTHORIZED,
+    HTTP_STATUS_FORBIDDEN,
+    HTTP_STATUS_NOT_FOUND,
+    HTTP_STATUS_INTERNAL_SERVER_ERROR,
+    HTTP_STATUS_BAD_GATEWAY,
+    HTTP_STATUS_SERVICE_UNAVAILABLE,
+    HTTP_STATUS_GATEWAY_TIMEOUT,
+],
+
+
+@pytest.fixture(scope="module")
+def service() -> tuple[str, str]:
+    """Deploys a Cloud Run service and returns its URL and a valid token."""
+    # Add a unique suffix to create distinct service names.
+    service_name = f"receive-{uuid.uuid4()}"
```

Comment:
You dropped the `.hex` from this call.
```suggestion
    service_name = f"receive-{uuid.uuid4().hex}"
```